TotalCMD for 32 and 64bit of TC

Novalis7747 opened the pull request:
### Link to issue number:
> Please include the issue number here. This helps us to keep the information linked together. If this is a minor/trivial change an issue does not need to be created. If in doubt, please create one.

### Summary of the issue:
> A quick summary of the problem you are trying to solve.
TotalCMD.py does handle only 32bit of TC and TotalCMD64.py is linked to totalcmd.py what doesn't work for TC 64bit
### Description of how this pull request fixes the issue:
totalcmd.py is updated to handle 64bit TC, it checks if 32bit or 64bit TC is running and uses the required classes and controls. 
I also added a file index to the end of each list item (x of x)

### Testing performed:

### Known issues with pull request:
> Any known issues or downsides of this approach. For instance: _Will not work with python 3_

### Change log entry:
> The section and description of this change to use for the changes file. Valid sections are:
> 
> * New features
- added a file index to the end of each list item (x of x)
> * Changes
> * Bug fixes
>
> For examples see the changes file in your NVDA repo: `user_docs/en/changes.t2t` or on github: https://github.com/nvaccess/nvda/blob/master/user_docs/en/changes.t2t

Novalis7747 committed:
now supports 32bit and 64bit version of TotalCMD

modified the existing code to work also for 64bit version of Total Commander

Novalis7747 committed:
Totalcmd.py: works now for 64bit of Total Commander and gives the file index (x of x) after the list item.

Totalcmd.py: works now for 64bit of Total Commander and gives the file index (x of x) after the list item.

Novalis7747 committed:
TotalCMD 32 and 64 bit capable

totalcmd.py handles now 64bit TC as well. Also added a file index to the list item output (i.e. 3 of 256)

Adriani90 commented:
I think commit 18c5e7df378534ef7c8c10f713ee114ed0ad91ea is subject to review. Is there any chance to include this in NVDA 2018.2? It does not affect translations as far as I can see.

Adriani90 commented:
@leonardder

Novalis7747 committed:
Total Commander 32/64 bit

reworked code so that it works for 32 and 64 bit of Total Commander
- fixed left/right announcements for ftp connections
- added hotkey ALT-1 to announce left/right and the current working directory or currently active Tab /if Tabs are used)

Novalis7747 committed:
Total Commander 32/64bit

reworked code to work with 32 and 64bit version of Total Commander
- fixed left right announcements for ftp connections
- added a file index (x of x)

Novalis7747 commented:
Latest commit should be used, fixed a bug that left/right wasn’t announced correctly when in a ftp connection….

From: Adriani90 <notifications@github.com>
Sent: Samstag, 19. Mai 2018 15:17
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Ralf Kefferpuetz <novalis7747@live.com>; Author <author@noreply.github.com>
Subject: Re: [nvaccess/nvda] TotalCMD for 32 and 64bit of TC (#8250)


@leonardder<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fleonardder&data=02%7C01%7C%7C349c6701f71e433579b508d5bd8ac96c%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636623326135797781&sdata=J%2FsPAOGDCDTi4diLyLnkkpsaRCUY%2FJ0hs7tfXu8G7As%3D&reserved=0>

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnvaccess%2Fnvda%2Fpull%2F8250%23issuecomment-390404317&data=02%7C01%7C%7C349c6701f71e433579b508d5bd8ac96c%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636623326135797781&sdata=NF%2BAg6N8HxdkQD18mxizZT%2F57zonefYkhrFgz5llpPY%3D&reserved=0>, or mute the thread<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAffwnNkVpfYCwKepS6VTi1Uwcgk4uST6ks5t0BtDgaJpZM4T2GKJ&data=02%7C01%7C%7C349c6701f71e433579b508d5bd8ac96c%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636623326135797781&sdata=s7niwUB%2FQ8OcVcBCVY6X57RgHBBIQRMn2t6e3ER1pe8%3D&reserved=0>.

leonardder commented:
This won't be in 2018.2, as the first release candidate is planned for 
the end of next week.

Novalis7747 committed:
Top/Bottom announcement and localized index

added Top / Bottom announcement for the beginning and end of the listview.
localized the index announcement

Novalis7747 committed:
solves issue #8244

solves #8244 , NVDA does not speak email contact autocompletion options in outlook 2016 and NVDA 2018.1
modified class:
class AutoCompleteListItem(IAccessible):

	def event_stateChange(self):
		states=self.states
		focus=api.getFocusObject()
		if (focus.role==controlTypes.ROLE_EDITABLETEXT or focus.role==controlTypes.ROLE_BUTTON) and controlTypes.STATE_SELECTED in states and controlTypes.STATE_INVISIBLE not in states and controlTypes.STATE_UNAVAILABLE not in states and controlTypes.STATE_OFFSCREEN not in states:
			address = self.name
			if address:
				speech.cancelSpeech()
				ui.message(self.name)

Novalis7747 committed:
fixes issue #8244

fixes issue #8244 , NVDA does not speak email contact autocompletion options in outlook 2016 and NVDA 2018.1

Adriani90 commented:
@Novalis7747 this could cause 
confusions for the final review of your total commander pull request. I suggest creating a separate branche and creating pull request from there. Me or @Joslee can assist you with that if needed.

Adriani90 commented:
CC: @leonardder. Maybe someone can review the last commit regarding total commander, this is commit
61af55f

Adriani90 commented:
and commit
e9e2975

derekriemer commented on source/appModules/outlook.py:
```diff
@@ -351,8 +351,10 @@ def event_stateChange(self):
+			address = self.name
+			if address:
+				speech.cancelSpeech()
+				ui.message(self.name).queue
```
I don't understand this line. ui.message returns None

derekriemer commented on source/appModules/totalcmd.py:
```diff
@@ -25,25 +32,32 @@ def event_gainFocus(self):
-			else:
-				ui.message(_("right"))
+			try:
+				if obj2.parent.parent.previous.firstChild.role  == 14:
```
Can you please use the constant from controlTypes.py?
you can import controlTypes, and then do something like == controlTypes.ROLE_LIST

derekriemer commented on source/appModules/totalcmd.py:
```diff
@@ -25,25 +32,32 @@ def event_gainFocus(self):
+				if obj2.parent.parent.previous.firstChild.role  == 14:
+					ui.message(_("left"))
+				else:
+					ui.message(_("right"))
```
see translators comment above

derekriemer commented on source/appModules/totalcmd.py:
```diff
@@ -25,25 +32,32 @@ def event_gainFocus(self):
 		if self.name:
+			currIndex = self.IAccessibleChildID
+			allIndex = self.parent.childCount
+			if currIndex == 1: ui.message(_("Top"))
```
please place on two lines, and the top message needs a translators comment.

derekriemer commented on source/appModules/totalcmd.py:
```diff
@@ -25,25 +32,32 @@ def event_gainFocus(self):
+			allIndex = self.parent.childCount
+			if currIndex == 1: ui.message(_("Top"))
+			if allIndex == currIndex: ui.message(_("Bottom"))
+			indexString=_("{number} of {total}").format( number = currIndex, total = allIndex)
```
Again we need a translators comment here.

derekriemer reviewed (changes requested):
Please see the pending review here.
Thanks.

Novalis7747 committed:
implemented reviewer comments

if statements splitted in 2 lines and translator lines defined.
At the reviewer and his first comment, that was for another issue regarding the outlook.py and came in here by mistake.
This is the TotalCmd.py only and has nothing to do with the outlook.py, just a wrong commit from my side, sorry.

Novalis7747 commented:
Hi Derek,

Thanks for reviewing.
All done.  Commit 4dc04cd
https://github.com/nvaccess/nvda/pull/8250/commits/4dc04cd3d76e3e1c38803e69836e8b7c99335d76

For your first comment, that’s not related to totalcmd. I committed it by mistake, it should go to a different issue for outlook.py.
The outlook issue is something different. Sorry again, Github kills me 😉

Regards,
   Ralf
From: Derek Riemer <notifications@github.com>
Sent: Donnerstag, 21. Juni 2018 12:15
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Ralf Kefferpuetz <novalis7747@live.com>; Mention <mention@noreply.github.com>
Subject: Re: [nvaccess/nvda] TotalCMD for 32 and 64bit of TC (#8250)


@derekriemer requested changes on this pull request.

Please see the pending review here.
Thanks.

________________________________

In source/appModules/outlook.py<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnvaccess%2Fnvda%2Fpull%2F8250%23discussion_r197078796&data=02%7C01%7C%7C4a32215432474c08f84c08d5d75fd414%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636651728926835945&sdata=F8k4AvESfPJt2uCfQWFoFmm%2F4rPU029UN67iKSEnLZ0%3D&reserved=0>:

> @@ -351,8 +351,10 @@ def event_stateChange(self):

               states=self.states

               focus=api.getFocusObject()

               if (focus.role==controlTypes.ROLE_EDITABLETEXT or focus.role==controlTypes.ROLE_BUTTON) and controlTypes.STATE_SELECTED in states and controlTypes.STATE_INVISIBLE not in states and controlTypes.STATE_UNAVAILABLE not in states and controlTypes.STATE_OFFSCREEN not in states:

-                      speech.cancelSpeech()

-                      ui.message(self.name)

+                      address = self.name

+                      if address:

+                              speech.cancelSpeech()

+                              ui.message(self.name).queue

I don't understand this line. ui.message returns None

________________________________

In source/appModules/totalcmd.py<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnvaccess%2Fnvda%2Fpull%2F8250%23discussion_r197079456&data=02%7C01%7C%7C4a32215432474c08f84c08d5d75fd414%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636651728926835945&sdata=ywZMRBSJLHLkriTO8SJAEZ6%2Fg4b4zv2mfPpZVYjHnr0%3D&reserved=0>:

>                              obj=obj.previous

-                              if obj.windowClassName!="TDrivePanel":

-                                     counter+=1

-                      if counter==2:

-                              ui.message(_("left"))

-                      else:

-                              ui.message(_("right"))

+                      try:

+                              if obj2.parent.parent.previous.firstChild.role  == 14:

Can you please use the constant from controlTypes.py?
you can import controlTypes, and then do something like == controlTypes.ROLE_LIST

________________________________

In source/appModules/totalcmd.py<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnvaccess%2Fnvda%2Fpull%2F8250%23discussion_r197080329&data=02%7C01%7C%7C4a32215432474c08f84c08d5d75fd414%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636651728926835945&sdata=CeWmUySzKITU1Kqfy%2B9dutJARDb1fTSGgg3EyAkXG5w%3D&reserved=0>:

>                              obj=obj.previous

-                              if obj.windowClassName!="TDrivePanel":

-                                     counter+=1

-                      if counter==2:

-                              ui.message(_("left"))

-                      else:

-                             ui.message(_("right"))

+                      try:

+                              if obj2.parent.parent.previous.firstChild.role  == 14:

+                                     ui.message(_("left"))

+                              else:

+                                     ui.message(_("right"))

see translators comment above

________________________________

In source/appModules/totalcmd.py<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnvaccess%2Fnvda%2Fpull%2F8250%23discussion_r197080744&data=02%7C01%7C%7C4a32215432474c08f84c08d5d75fd414%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636651728926835945&sdata=UxhI2Y2%2Bmd49fXBhjFGMDnrgjxdpjuzJAa5LD8kuLBU%3D&reserved=0>:

>              super(TCList,self).event_gainFocus()



        def reportFocus(self):

               if self.name:

+                      currIndex = self.IAccessibleChildID

+                      allIndex = self.parent.childCount

+                      if currIndex == 1: ui.message(_("Top"))

please place on two lines, and the top message needs a translators comment.

________________________________

In source/appModules/totalcmd.py<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnvaccess%2Fnvda%2Fpull%2F8250%23discussion_r197080836&data=02%7C01%7C%7C4a32215432474c08f84c08d5d75fd414%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636651728926835945&sdata=iobCY3vUWnCvMOwxHgeL61RgiUVi1%2B9W%2F2ODt1poONk%3D&reserved=0>:

>              super(TCList,self).event_gainFocus()



        def reportFocus(self):

               if self.name:

+                      currIndex = self.IAccessibleChildID

+                      allIndex = self.parent.childCount

+                      if currIndex == 1: ui.message(_("Top"))

+                      if allIndex == currIndex: ui.message(_("Bottom"))

+                      indexString=_("{number} of {total}").format( number = currIndex, total = allIndex)

Again we need a translators comment here.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnvaccess%2Fnvda%2Fpull%2F8250%23pullrequestreview-130748779&data=02%7C01%7C%7C4a32215432474c08f84c08d5d75fd414%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636651728926835945&sdata=zEGaUMFIbIHB%2F5Zj3jajCCrMN5ByNTymaAW6IoAJV8s%3D&reserved=0>, or mute the thread<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAffwnBRR0arJEgRo5BtB0GBTVifqyZhkks5t-3IcgaJpZM4T2GKJ&data=02%7C01%7C%7C4a32215432474c08f84c08d5d75fd414%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636651728926835945&sdata=mbhBJgrou8l%2F2BUTeyVOuue1SSpmghbB83H%2BkpmsPOQ%3D&reserved=0>.

Novalis7747 commented:
this outlook fix came in as a commit and I can't delete commits. The only way to do that is to delete the reposity according to the github manual.
May I ask you to close this PR please?
I need to start over with Github and will reading manuals first. When I've  sorted my mess out I'm going to raise a new PR.

thanks

derekriemer commented on source/appModules/totalcmd.py:
```diff
@@ -50,8 +52,13 @@ def reportFocus(self):
+			if allIndex == currIndex:
+				# Translators: the word  Bottom for reaching the Bottom of the list in your language (only the word Bottom).
+				ui.message(_("Bottom"))
+			# Translators: the phrase "{number} of {total}" in an index like 1 of 32 in your language.
```
NIT: I think in an index sounds better as is an index

derekriemer commented on source/appModules/totalcmd.py:
```diff
@@ -38,9 +38,11 @@ def event_gainFocus(self):
 			try:
-				if obj2.parent.parent.previous.firstChild.role  == 14:
+				if obj2.parent.parent.previous.firstChild.role  == controlTypes.ROLE_LIST:
+					# Translators: the word left for the left window in your language (only the word left).
```
To clarify, do you mean "a message telling the user that they reached the left most column in this window" or something similar?

derekriemer reviewed (commented):
This is about ready in my opinion although a core developer needs to approve it.

josephsl commented:
Hi,

Taking this PR offline as requested by feature author. Thanks.
